scripts/import-cp-contexts.sh

```diff
@@ -0,0 +1,247 @@
+--replace-localhost <address> replaces server addresses "127.0.0.1" with <address>
+--merge                       merge the control planes contexts with the existing cluster contexts
+-o <filename>                 specify a different kubeconfig file to save the contexts (- for stdout)
+--silent                      no information output
+-X                            enable verbose execution of the script for debugging
+EOF
+}
+
+
+get_kubeconfig() {
+    context="$1"
+    cp_name="$2"
+    cp_type="$3"
+    ip_addr="$4"
+
+    echov "Getting the kubeconfig of Control Plane \"$cp_name\" of type \"$cp_type\" from context \"$context\":"
+
+    # wait for CP ready
+    while [[ $(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context get controlplane $cp_name -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True" ]]; do
+        echov "* Waiting for Control Plane \"$cp_name\" in context \"$context\" to be Ready..."
+        sleep 5
+    done
+
+    # put into the shell variable "kubeconfig" the kubeconfig contents for use from outside of the hosting cluster
+    if [[ "$cp_type" == "host" ]] ; then
+        echov "- Using hosting cluster context \"${context}\""
+        kubeconfig=$(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context config view --flatten --minify)
+    else
+        # determine the secret name and namespace
+                    key=$(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context get controlplane $cp_name -o=jsonpath='{.status.secretRef.key}')
+            secret_name=$(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context get controlplane $cp_name -o=jsonpath='{.status.secretRef.name}')
+        secret_namespace=$(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context get controlplane $cp_name -o=jsonpath='{.status.secretRef.namespace}')
+        # get the kubeconfig
+        echov "- Using \"$key\" from \"$secret_name\" secret in \"$secret_namespace\""
+        kubeconfig=$(KUBECONFIG="$in_KUBECONFIG" kubectl --context $context get secret $secret_name -n $secret_namespace -o=jsonpath="{.data.$key}" | base64 -d)
+    fi
+
+    # make kubeconfig unique for the control plane
+    echov "- Making the kubeconfig unique..."
+    kubeconfig=$(
+        echo "$kubeconfig" \
+        | yq ".clusters[0].name = \"$cp_name-cluster\"" \
+        | yq ".users[0].name = \"$cp_name-admin\"" \
+        | yq ".contexts[0].name = \"$cp_name\"" \
+        | yq ".contexts[0].context.cluster = \"$cp_name-cluster\"" \
+        | yq ".contexts[0].context.user = \"$cp_name-admin\"" \
+        | yq ".current-context = \"$cp_name\""
+    )
+
+    # swap out 127.0.0.1 with an external ip address
+    if [[ "$ip_addr" != "" ]] ; then
+        echov "- Replacing server ip address \"127.0.0.1\" with \"$ip_addr\""
+        kubeconfig=$(echo "$kubeconfig" | sed "s@server: https://127\.0\.0\.1:@server: https://$ip_addr:@g")
+    fi
+
+    # return the kubecconfig of the control plane in plain text
```

Comment:
What is "in plain text" trying to say here? Is this about the difference between file name and file contents? If so then a clearer comment might be something like "Return the kubeconfig contents" or "Return the kubeconfig file contents" or "Return the kubeconfig YAML".

Reply:
ok reworded